File: src/main/java/example/sort/tree/Node.java
package example.sort.tree;

import otterop.lang.Array;
import otterop.lang.MakePure;

@MakePure
public class Node {

    private Array<Node> _children;

    private int _value;

    public Node(int value, Array<Node> children) {
        this._value = value;
        this._children = children;
    }

    public int value() {
        return this._value;
    }

    public Node child(int i) {
        Array<Node> children = this._children;
        return children.get(i);
    }

    public Array<Node> children() {
        return this._children;
    }
}

File: dotnet/Example/Sort/Tree/Node.cs
namespace Example.Sort.Tree
{
    public class Node
    {
        private Otterop.Lang.Array<Node> _children;
        private int _value;
        public Node(int value, Otterop.Lang.Array<Node> children)
        {
            this._value = value;
            this._children = children;
        }

        public int Value()
        {
            return this._value;
        }

        public Node Child(int i)
        {
            Otterop.Lang.Array<Node> children = this._children;
            return children.Get(i);
        }

        public Otterop.Lang.Array<Node> Children()
        {
            return this._children;
        }

    }

}
